test/edu/csus/ecs/pc2/core/UtilitiesTest.java

```diff
@@ -516,4 +516,107 @@ public void testgetCDPDataDirectories() throws Exception {
         
     }
     
+    /**
+     * Test where both arrays are null.
+     * @throws Exception
+     */
+    public void testBothNull() throws Exception {
+        String[] one = null;
+        String[] two = null;
+
+        String[] newArray = Utilities.copyArray(one, two);
+
+        assertEquals("Expecting a zero lenght array", 0, newArray.length);
+
+    }
+
+    /**
+     * Test where first array is null.
+     * 
+     * @throws Exception
+     */
+    public void testCopyEmptyOneNull() throws Exception {
+
+        String[] one = null;
+        String[] two = { "omega", "delta" };
+
+        String[] newArray = Utilities.copyArray(one, two);
+
+        if (one == null) {
+            one = new String[0];
+        }
+
+        assertEquals("Expect result array length", one.length + two.length, newArray.length);
+
+    }
+
+    /**
+     * Test where second array is null.
+     * 
+     * @throws Exception
+     */
+    public void testCopyEmptyTwoNull() throws Exception {
+
+        String[] one = { "alpha", "beta", "gamma" };
+        String[] two = null;
+
+        String[] newArray = Utilities.copyArray(one, two);
+
+        if (two == null) {
+            two = new String[0];
+        }
+
+        assertEquals("Expect result array length", one.length + two.length, newArray.length);
+    }
+
+    /**
+     * Test copy where second array is zero length.
+     * @throws Exception
+     */
+    public void testCopyEmptyTwo() throws Exception {
+
+        String[] one = { "alpha", "beta", "gamma" };
+        String[] two = new String[0];
+
+        String[] newArray = Utilities.copyArray(one, two);
+        assertEquals("Expect result array length", one.length + two.length, newArray.length);
+    }
+
+    /**
+     * Test copy non-empty arrays.
+     * 
+     * @throws Exception
+     */
+    public void testCopyArrays() throws Exception {
+
+        String[] one = { "alpha", "beta", "gamma" };
+        String[] two = { "omega", "delta" };
+
+        String[] newArray = Utilities.copyArray(one, two);
+        assertEquals("Expect result array length", one.length + two.length, newArray.length);
+
+        for (int i = 0; i < one.length; i++) {
+            int naOffset = i + 0;
+            assertEquals("Expected element equal one[" + i + "]", one[i], newArray[naOffset]);
+        }
+
+        for (int i = 0; i < two.length; i++) {
+            int naOffset = i + one.length;
+
+            assertEquals("Expected element equal two[" + i + "]", two[i], newArray[naOffset]);
+        }
+
+    }
+
+    public void testCopy() throws Exception {
```

Comment:
I don't necessarily object to it, but isn't this method testing nothing but the Java library (specifically, `System.arraycopy()`)?  Seems like it's not useful to spending time testing Java itself (how many thousands of other Java library methods are we also using, but NOT testing?)

Reply:
This was tested in order to confirm that the library routine worked as described.  It was used to
confirm an understanding of how arrayCopy worked.   And if it fails then other methods that
use that method might fail.